spec/spec.md

```diff
@@ -47,6 +47,9 @@ innermost active invocation of [Executing a File](#executing-a-file).
 
 [configuration]: modules.md#configuration
 
+ > Each new *configuration* has a unique ID by default. Create a new
+ > configuration with the same ID to share identities.
```

Comment:
Block quotes are used for non-normative information—things that help the reader understand _why_ the spec works the way it does, or clarifies an interaction that's implicit in the normative spec text. They shouldn't be used to to specify how Sass behaves.

I don't think this is the right place to specify when configurations do or don't share IDs, either. The "Current Configuration" definition explains which configuration object is current, but it's not related to the process of creating configurations. I think instead you want to say in the definition of "Configuration":

> A new configuration has a unique ID unless otherwise specified.

...and then in the semantics for the `@forward` rule:

> ...an empty configuration with the same ID as the current configuration.

You also need to specify what to _do_ with the IDs. Currently the "Loading a Module" procedure still says "If `config` is not empty, throw an error", but that's no longer what we want.

Reply:
I think I know what you mean here and on the comment below.. it may be working that way on the implementation but it's definitely not clear in here, let me update this and double check the impl actually does this :)

Reply:
I have updated the PR but I didn't use this behavior:
> an empty configuration with the same ID as the current configuration.

Instead I used:
> The `rule-config` opaque ID will be generated deterministically for the location it was declared at.

I guess the former behavior would decouple the config ID generation from the file location 🤔 and I just tested it and the example you shared below (but changing `$var2: from downstream 1 !default`) worked

I found the implementation to be slightly more complex (having to carry the opaqueId around) but I guess it works.. so let me re-update the PR

Reply:
Done.